homeassistant/components/linksys_smart/device_tracker.py

```diff
@@ -68,15 +68,21 @@ def _update_info(self):
             return False
         try:
             data = response.json()
-            result = data["responses"][0]
-            devices = result["output"]["devices"]
+            devices = data["responses"][0]["output"]["devices"]
+            connections = data["responses"][1]["output"]["connections"]
             for device in devices:
                 if not (macs := device["knownMACAddresses"]):
                     _LOGGER.warning("Skipping device without known MAC address")
                     continue
                 mac = macs[-1]
-                if not device["connections"]:
-                    _LOGGER.debug("Device %s is not connected", mac)
+                device_found = False
+                for mac_address in macs:
+                    if any(conn["macAddress"] == mac_address for conn in connections):
+                        device_found = True
+                        break
+
+                if not device_found:
+                    _LOGGER.debug("Device %s is not connected", mac_address)
```

Comment:
This integration/code needs to be extracted into its own PyPi package first before we start making changes to the integration.